pkg/drain/nodeDrainStrategy.go

```diff
@@ -49,18 +59,40 @@ func (ds *osdDrainStrategy) Execute(node *corev1.Node, logger logr.Logger) ([]*D
 			return nil, fmt.Errorf("cannot determine drain commencement time for node %v", node.Name)
 		}
 		me := &multierror.Error{}
-		for _, ds := range ds.timedDrainStrategies {
-			dsName := ds.GetName()
-			expectedTime := result.AddedAt.Add(ds.GetWaitDuration())
+		for _, tds := range ds.timedDrainStrategies {
+			dsName := tds.GetName()
+			expectedTime := result.AddedAt.Add(tds.GetWaitDuration())
 			drainStrategyMsg := fmt.Sprintf("drain strategy %v for node %v, commencing drain at %v, execution expected after %v", dsName, node.Name, result.AddedAt, expectedTime)
-			if isAfter(result.AddedAt, ds.GetWaitDuration()) {
+			if isAfter(result.AddedAt, tds.GetWaitDuration()) {
 				logger.Info(fmt.Sprintf("Executing %s", drainStrategyMsg))
-				r, err := ds.GetStrategy().Execute(node, logger)
+				r, err := tds.GetStrategy().Execute(node, logger)
 				if err != nil {
 					return nil, err
 				}
 				me = multierror.Append(err, me)
 				if r.HasExecuted {
+					if dsName == pdbPodDeleteName {
+						// Check if a notification for it has been sent successfully - if so, nothing to do
+						isNotified, err := ds.metricsClient.IsMetricNotificationEventSentSet(ds.uc.Name,
+							string(notifier.MuoStateDelayed), ds.uc.Spec.Desired.Version)
+						if err != nil {
+							logger.Error(err, "Failed to send the service log about upgrade delay due to node drain grace period")
+							return nil, fmt.Errorf("can't check cluster metric NotificationSent: %v", err)
+						}
+						if !isNotified {
+							logger.Info("Sending upgrade delay message about node drain grace period")
+							msg := "Node drain grace period might be impacting cluster upgrade. " +
+								"Please refer to the article for further details https://access.redhat.com/solutions/7075425"
```

Comment:
```suggestion
							msg := "Cluster upgrade is experiencing delay due to node drain grace period. " + "Please refer to the section on 'PDBHealthcheckFailed' for further details in https://access.redhat.com/solutions/7075425"
```

Reply:
Overall, I am -1 to have KCS links and notification content in MUO code but for time being it's fine. We'll need a better way to handle that aspect across the packages eventually.